l2geth/miner/worker.go

```diff
@@ -196,7 +196,7 @@ func newWorker(config *Config, chainConfig *params.ChainConfig, engine consensus
 		unconfirmed:        newUnconfirmedBlocks(eth.BlockChain(), miningLogAtDepth),
 		pendingTasks:       make(map[common.Hash]*task),
 		txsCh:              make(chan core.NewTxsEvent, txChanSize),
-		rollupCh:           make(chan core.NewTxsEvent, txChanSize),
+		rollupCh:           make(chan core.NewTxsEvent, 1),
```

Comment:
does this change functionality so that we only process 1 tx at a time?

Reply:
Its only possible to really process 1 transaction at a time due to the single threaded nature of evm execution

Reply:
The change in functionality here is reducing the buffer size of the channel which will cause writing to the channel to block when its full. Technically the updated recovery logic should allow us to revert this change. This change will add a slight performance hit